lib/uri/common.rb

```diff
@@ -88,9 +102,21 @@ def self.scheme_list
   private_constant :INITIAL_SCHEMES
   Ractor.make_shareable(INITIAL_SCHEMES) if defined?(Ractor)
 
+  # Returns a new object constructed from the given +scheme+, +arguments+,
+  # and +default+:
+  #
+  # - The new object is a instance <tt>URI.scheme_list[scheme.upcase]</tt>.
+  # - The object is initialized by calling the class initializer
+  #   using +scheme+ and +arguments+.
+  #   See URI::Generic.new.
   #
-  # Construct a URI instance, using the scheme to detect the appropriate class
-  # from +URI.scheme_list+.
+  # Examples:
+  #
+  #   values = ['john.doe', 'www.example.com', '123', nil, '/forum/questions/', nil, 'tag=networking&order=newest', 'top']
+  #   URI.for('https', *values)
+  #   # => #<URI::HTTPS https://john.doe@www.example.com:123/forum/questions/?tag=networking&order=newest#top>
+  #   URI.for('foo', *values, URI::HTTP)
+  #   # => #<URI::Generic foo://john.doe@www.example.com:123/forum/questions/?tag=networking&order=newest#top>
```

Comment:
This example isn't using the `default` passed in since it was passed in as a positional argument rather than a keyword argument. Did you mean to do something like this?

```suggestion
  #   URI.for('foo', *values, default: URI::HTTP)
  #   # => #<URI::HTTP foo://john.doe@www.example.com:123/forum/questions/?tag=networking&order=newest#top>
```